From 088f793ebe0403438ceab9a993405ab8a9bfc183 Mon Sep 17 00:00:00 2001
From: Robert Marshall <rmarshall@redhat.com>
Date: Fri, 29 Jan 2016 17:34:02 -0500
Subject: [PATCH 77/97] Fix locale issue in grub-setpassword (#1294243)

A shell substitution was expecting non-translated output to grab the
hashed password and put it in the user.cfg file. Modified code to force
the generic C locale when this particular piece of code is run.

Resolves: rhbz#1294243
---
 util/grub-setpassword.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/grub-setpassword.in b/util/grub-setpassword.in
index dd76f00fc0e5..2923f437ebe8 100644
--- a/util/grub-setpassword.in
+++ b/util/grub-setpassword.in
@@ -105,7 +105,7 @@ getpass() {
     P1="$1" && shift
 
     ( echo ${P0} ; echo ${P1} ) | \
-        ${grub_mkpasswd} | \
+        LC_ALL=C ${grub_mkpasswd} | \
         grep -v '[eE]nter password:' | \
         sed -e "s/PBKDF2 hash of your password is //"
 }
-- 
2.13.0

